client/state/sites/terms/reducer.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import merge from 'lodash/merge';
+import uniq from 'lodash/uniq';
+import get from 'lodash/get';
+import keyBy from 'lodash/keyBy';
+
+/**
+ * Internal dependencies
+ */
+import {
+	TERMS_RECEIVE,
+	DESERIALIZE
+} from 'state/action-types';
+import { isValidStateWithSchema } from 'state/utils';
+import {
+	termsSchema,
+	taxonomiesSchema
+} from './schema';
+
+export function taxonomies( state = {}, action ) {
```

Comment:
Not sure I'm following your proposed change here.  The addition being an additional object layer called `items`?